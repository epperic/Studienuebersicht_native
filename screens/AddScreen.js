import React from 'react';
import Form from '../components/Form';
import { LoadingCircle } from '../components/LoadingCircle';
import firebase from '../database/firebaseDB';


class AddScreen extends React.Component {
  constructor(props) {
    super(props);
    this.dbRef = firebase.firestore().collection('modul');
    this.state = {
      isLoading: false,
    };
    this.firestoreSave = this.firestoreSave.bind(this);
  }

  firestoreSave(modul) {
    this.setState({
      isLoading: true,
    });

    this.dbRef.add(modul).then((res) => {
      this.setState({
        isLoading: false,
      });
      console.log('Modul "' + modul.name + '" wurde erfolgreich hinzugefügt.');
      this.props.navigation.navigate('BottomNavigation', { screen: 'Semesters' });
    })
      .catch((err) => {
        console.error("Fehler beim Hinzufügen eines Moduls in Firestore: ", err);
        this.setState({
          isLoading: false,
        });
      });
  }

  render() {
    return !this.state.isLoading ? <Form firestoreSave={this.firestoreSave} /> : <LoadingCircle />
  }
}

export default AddScreen;