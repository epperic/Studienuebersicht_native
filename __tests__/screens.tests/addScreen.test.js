
import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import AddScreen from '../../screens/AddScreen';
jest.useFakeTimers()

describe('Addscreen', () => {
  it('renders correctly', () => {
    const addScreen = renderer.create(
      <AddScreen />
    ).toJSON();
    expect(addScreen).toMatchSnapshot();
  });
});