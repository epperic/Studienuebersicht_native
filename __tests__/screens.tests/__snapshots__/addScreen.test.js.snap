// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addscreen renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 15,
      "marginTop": 32,
      "paddingHorizontal": 24,
    }
  }
>
  <View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Modulname
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Professor
      </Text>
      <TextInput
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        ECTS
      </Text>
      <TextInput
        keyboardType="numeric"
        maxLength={2}
        onChangeText={[Function]}
        placeholder="1-30 "
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Note
      </Text>
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="z.B 1.7 (Punkttrennung)"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
          }
        }
        value=""
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Semester
      </Text>
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 35,
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "green",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": 64,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#ffffff",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "lineHeight": 16,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰄬
              </Text>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Modul hinzufügen
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
